src/main/java/TaskManager.java

```diff
@@ -0,0 +1,162 @@
+                throw new IllegalSavingAction();
+            }
+        }
+
+    }
+    /**
+     * Adds a general task in the list
+     * @param name refers to the name of the task
+     */
+    public void addTask(String name) {
+        Task newTask = new Task(name);
+        Tasks.add(newTask);
+        chatbox.setContent("added: " + name);
+        chatbox.chatboxPrinter();
+    }
+
+    /**
+     * Adds a deadline task to the list
+     * @param name refers to the name of the task
+     * @param by refers to the deadline of the task
+     */
+    public void addDeadline(String name, String by) {
+        Deadline newDeadline = new Deadline(name, by);
+        Tasks.add(newDeadline);
+        int s = Tasks.size();
+        String response = ADD_RES + newDeadline.getListName() +
+                "\n" + "Now you have " + String.valueOf(s) + " tasks in your list.";
+        chatbox.setContent(response);
+        chatbox.chatboxPrinter();
+    }
+
+    /**
+     * Adds an event task to the list
+     * @param name refers to the name of the task
+     * @param at refers to the happening time of the event
+     */
+    public void addEvent(String name, String at) {
+        Event newEvent = new Event(name, at);
+        Tasks.add(newEvent);
+        int s = Tasks.size();
+        String response = ADD_RES + newEvent.getListName() +
+                "\n" + "Now you have " + String.valueOf(s) + " tasks in your list.";
+        chatbox.setContent(response);
+        chatbox.chatboxPrinter();
+    }
+
+    /**
+     * Adds a todo task to the list
+     * @param name refers to the name of the todo task
+     */
+    public void addToDo(String name) {
+        ToDo newToDo = new ToDo(name);
+        Tasks.add(newToDo);
+        int s = Tasks.size();
+        String response = ADD_RES + newToDo.getListName() + "\n" +
+                "Now you have " + String.valueOf(s) + " tasks in your list.";
+        chatbox.setContent(response);
+        chatbox.chatboxPrinter();
+    }
+
+    /**
+     * Lists all tasks in the list with adding time order
+     */
+    public void listTask() {
+        String content = "Here are the tasks in your list:\n";
+        if(Tasks.size() == 0){
+            content = "Sorry, there's no task in the list :(";
+        }else {
+            for (int i = 0; i < Tasks.size(); i++) {
+                String index = String.valueOf(i + 1);
+                String name = index + ". " + Tasks.get(i).getListName();
+                content += name;
+                if(i < Tasks.size() - 1){
+                    content += "\n";
+                }
+            }
+        }
+        chatbox.setContent(content);
+        chatbox.chatboxPrinter();
+    }
+
+    /**
+     * Marks specific task in the list as done
+     * @param n refers to the index of the task in adding time order
+     */
+    public void markTask(int n) {
+        System.out.println(n);
+        String content = "";
+        if(n < 1 || n > Tasks.size()) {
```

Comment:
This is a compound boolean expression which you can further refactor. See [sample boolean expressions](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions).